cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Util.hs

```diff
@@ -0,0 +1,135 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cardano.DbSync.Era.Shelley.Util
+  ( blockHash
+  , blockNumber
+  , blockPrevHash
+  , blockSize
+  , blockTxCount
+  , blockTxs
+  , epochNumber
+  , fakeGenesisHash
+  , mkSlotLeader
+  , pointToSlotHash
+  , renderHash
+  , slotNumber
+  , txFee
+  , txHash
+  , txInputList
+  , txOutputList
+  , txOutputSum
+  , unCoin
+  , unHeaderHash
+  ) where
+
+import           Cardano.Prelude
+
+import           Cardano.Chain.Slotting (SlotNumber (..))
+import qualified Cardano.Crypto.Hash as Crypto
+
+import qualified Cardano.Db as Db
+import           Cardano.DbSync.Types
+
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.ByteString.Char8 as BS
+import           Data.Sequence.Strict (StrictSeq (..))
+import qualified Data.Text.Encoding as Text
+
+import qualified Ouroboros.Consensus.Shelley.Ledger.Block as Shelley
+import           Ouroboros.Network.Block (BlockNo (..), Point (..), SlotNo (..))
+import           Ouroboros.Network.Point (WithOrigin (..))
+import qualified Ouroboros.Network.Point as Point
+
+import           Shelley.Spec.Ledger.Coin (Coin (..))
+import qualified Shelley.Spec.Ledger.BlockChain as Shelley
+import qualified Shelley.Spec.Ledger.Tx as Shelley
+
+
+blockHash :: ShelleyBlock -> ByteString
+blockHash = unHeaderHash . Shelley.shelleyBlockHeaderHash
+
+blockNumber :: ShelleyBlock -> Word64
+blockNumber =
+  unBlockNo . Shelley.bheaderBlockNo . Shelley.bhbody . Shelley.bheader . Shelley.shelleyBlockRaw
+
+blockPrevHash :: ShelleyBlock -> ByteString
+blockPrevHash blk =
+  case Shelley.bheaderPrev (Shelley.bhbody . Shelley.bheader $ Shelley.shelleyBlockRaw blk) of
+    Shelley.GenesisHash -> fakeGenesisHash
+    Shelley.BlockHash h -> Crypto.getHash $ Shelley.unHashHeader h
+
+blockSize :: ShelleyBlock -> Word64
+blockSize = fromIntegral . Shelley.bBodySize . Shelley.bbody . Shelley.shelleyBlockRaw
+
+blockTxCount :: ShelleyBlock -> Word64
+blockTxCount = fromIntegral . length . getTxInternalUnsafe . Shelley.bbody . Shelley.shelleyBlockRaw
+
+blockTxs :: ShelleyBlock -> [ShelleyTx]
+blockTxs =
+    txList . Shelley.bbody . Shelley.shelleyBlockRaw
+  where
+    txList :: ShelleyTxSeq -> [ShelleyTx]
+    txList (Shelley.TxSeq txSeq) = toList txSeq
+
+epochNumber :: ShelleyBlock -> Word64 -> Word64
+epochNumber blk slotsPerEpoch = slotNumber blk `div` slotsPerEpoch
+
+fakeGenesisHash :: ByteString
+fakeGenesisHash = BS.take 32 (" G e n e s i s - H a s h " <> BS.replicate 32 ' ')
```

Comment:
Just like that?

Reply:
Placeholder because we need the correct thing in `ledger-specs`.